src/strgen/strgen_base.cpp

```diff
@@ -550,91 +479,96 @@ static bool CheckCommandsMatch(const char *a, const char *b, const char *name)
 	return result;
 }
 
-void StringReader::HandleString(char *str)
+[[nodiscard]] static std::string_view StripTrailingWhitespace(std::string_view str)
 {
-	if (*str == '#') {
-		if (str[1] == '#' && str[2] != '#') this->HandlePragma(str + 2, _strgen.lang);
-		return;
+	auto len = str.find_last_not_of("\r\n ");
+	if (len == std::string_view::npos) {
+		len = 0;
+	} else {
+		++len;
 	}
+	return str.substr(0, len);
```

Comment:
```suggestion
	if (len == std::string_view::npos) return {};
	return str.substr(0, len + 1);
```
A potential simpler variant, or is it important that `data()` still points to `str`?

Reply:
No, `data` is not important.